tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php

```diff
@@ -229,6 +231,94 @@ public function testClassWithReturnTypesOnProxiedMethods()
         $this->assertEquals(1, substr_count($classCode, 'function returnsParent(): \stdClass'));
     }
 
+    public function testClassWithNullableTypeHintsOnProxiedMethods()
+    {
+        if (PHP_VERSION_ID < 70100) {
+            $this->markTestSkipped('Nullable type hints are only supported in PHP >= 7.1.0.');
+        }
+
+        if (!class_exists('Doctrine\Tests\Common\ProxyProxy\__CG__\NullableTypeHintsClass', false)) {
+            $className = NullableTypeHintsClass::class;
+            $metadata = $this->createClassMetadata($className, ['id']);
+
+            $proxyGenerator = new ProxyGenerator(__DIR__ . '/generated', __NAMESPACE__ . 'Proxy', true);
+            $this->generateAndRequire($proxyGenerator, $metadata);
+        }
+
+        $classCode = file_get_contents(__DIR__ . '/generated/__CG__DoctrineTestsCommonProxyNullableTypeHintsClass.php');
+
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintInt(?int $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintObject(?\stdClass $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintSelf(?\\' . $className . ' $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefault(?int $param = 123)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefaultNull(?int $param = NULL)'));
```

Comment:
@trowski That's a good question, probably not, but then someone could use 7.1 on CI server to warm-up cache (and generate proxies) and then 7.0 in production...

I already added special case for null default value, but then it has opposite effect.
I actually thought that this code would throw an error due to incompatible signatures:
```php
class Foo {
	function x(?int $x = NULL) {}
}
class Bar extends Foo {
	function x(int $x = NULL) {}
}
```
But surprisingly it doesn't, even though subclass removes nullability (question mark) from the type, something I'd not really expect. This is super-confusing, really something not well-thought-out in my opinion... 😳